File: arrays/easy/remove_duplicates_from_sorted_array.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        int p1 = 0;
        int p2 = 1;
        int n  = nums.size();
        if(n==0){
            return 0;
        }
        for(int i=1;i<n;i++){
            if(nums[p1]!=nums[p2]){
                nums[p1+1] = nums[p2];
                p1++;
            }
            p2++;
        }
        return p1+1;


    }
};

File: arrays/easy/remove_duplicates_from_sorted_array.py
from typing import List


class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:

        next_non_duplicate = 1
        i = 1
        while i < len(nums):
            if nums[i] != nums[next_non_duplicate - 1]:
                nums[next_non_duplicate] = nums[i]
                next_non_duplicate += 1
            i += 1
        return next_non_duplicate